pkg/middleware/validate_action_url.go

```diff
@@ -0,0 +1,83 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+func ValidateActionUrl(cfg *setting.Cfg) web.Handler {
+	return func(c *contextmodel.ReqContext) {
+		// ignore local render calls
+		if c.IsRenderCall {
+			return
+		}
+
+		// only process requests targeting local instance
+		if !isLocalPath(c) {
+			// call next on url?
```

Comment:
I believe we agreed that non-local URLs are out of scope for blocking right now, so I think it'd be fine to call next here and let it through.